ui/src/app/applications/components/application-node-info/application-node-info.tsx

```diff
@@ -167,8 +166,46 @@ export const ApplicationNodeInfo = (props: {
         });
     }
 
+    const getPodReadinessGatesState = React.useCallback(_getPodReadinessGatesState, [props.live]);
+
+    let ReadinessGatesFailedWarning: React.ReactNode = null;
+    if (props.live && props.node.kind === 'Pod') {
+        const readinessGatesState = getPodReadinessGatesState(props.live);
+
+        if (readinessGatesState.failedConditions.length > 0 || readinessGatesState.nonExistConditions.length > 0) {
```

Comment:
These should have null protection, i.e. `(readinessGatesState?.failedConditions || []).length`

Reply:
the return of function `getPodReadinessGatesState` is surely an object with given keys and empty array as default